tests/test-upgrade.py

```diff
@@ -115,8 +122,18 @@ def test_upgrade(self):
             print("Testing {}".format(test))
             validation()
 
-        # On lxc umount docker overlay is not permitted.
-        try:
-            check_call("sudo grep -E (lxc|hypervisor) /proc/1/environ /proc/cpuinfo".split())
-        except CalledProcessError:
+        if not in_virtual_env():
+            # On lxc umount docker overlay is not permitted.
             check_call("sudo snap remove microk8s".split())
+
+
+def in_virtual_env():
+    '''
+    Returns: True if the deployment is in a VM/container.
+
+    '''
+    try:
+        check_call("sudo grep -E (lxc|hypervisor) /proc/1/environ /proc/cpuinfo".split())
+        return True
```

Comment:
Do we want to return True for all virt environments?  ie, VMs like kvm, qemu, vmware in addition to containers like lxc, docker, and rkt?

We may be better off using `systemd-detect-virt` instead of relying on `/proc`:
https://www.freedesktop.org/software/systemd/man/systemd-detect-virt.html

You can call it to detect *all* virt like this (rc = 0 means it's a virtual env):
```
subprocess.call(['systemd-detect-virt']) == 0
```

Or just containers like this:
```
subprocess.call(['systemd-detect-virt', '--container']) == 0
```